libbeat/docs/monitoring/monitoring-metricbeat.asciidoc

```diff
@@ -0,0 +1,268 @@
+http.port: 5067
+----------------------------------
+// end::enable-http-endpoint[]
+--
+
+. Disable the default collection of {beatname_uc} monitoring metrics. +
++
+--
+// tag::disable-beat-collection[]
+Add the following setting in the {beatname_uc} configuration file
+(+{beatname_lc}.yml+): 
+
+[source,yaml]
+----------------------------------
+monitoring.enabled: false
+----------------------------------
+// end::disable-beat-collection[]
+
+For more information, see 
+<<configuration-monitor,Monitoring configuration options>>.
+--
+
+ifndef::serverless[]
+. <<{beatname_lc}-starting,Start {beatname_uc}>>.
+endif::serverless[]
+
+[float]
+[[configure-metricbeat]]
+=== Install and configure {metricbeat} to collect monitoring data
+
+ifeval::["{beatname_lc}"!="metricbeat"]
+. {metricbeat-ref}/metricbeat-installation.html[Install {metricbeat}] on the
+same server as {beatname_uc}. If you already have {metricbeat} installed on the
+server, skip this step.
+endif::[]
+ifeval::["{beatname_lc}"=="metricbeat"]
+. The next step depends on how you want to run {metricbeat}:
+* If you're running as a service and want to run a separate monitoring instance,
+take the the steps required for your environment to run two instances of
+{metricbeat} as a service. The steps for doing this vary by platform and are
+beyond the scope of this documentation.
+* If you're running the binary directly in the foreground and want to run a
+separate monitoring instance,
+{metricbeat-ref}/metricbeat-installation.html[install {metricbeat}] to a
+different path. If necessary, set `path.config`, `path.data`, and `path.log`
+to point to the correct directories. See <<directory-layout>> for the default
+locations.
+* If you're using {metricbeat} to monitor itself, stop {metricbeat} and apply
+the following steps to your existing installation.
+endif::[]
+
+. Enable the beat-xpack module in {metricbeat}. +
+endif::[]
++
+--
+// tag::enable-beat-module[]
+For example, to enable the default configuration in the `modules.d` directory, 
+run the following command, using the correct command syntax for your OS:
+
+["source","sh",subs="attributes,callouts"]
+----------------------------------------------------------------------
+metricbeat modules enable beat-xpack
+----------------------------------------------------------------------
+
+For more information, see 
+{metricbeat-ref}/configuration-metricbeat.html[Specify which modules to run] and 
+{metricbeat-ref}/metricbeat-module-beat.html[beat module]. 
+// end::enable-beat-module[]
+--
+
+. Configure the beat-xpack module in {metricbeat}. +
++
+--
+// tag::configure-beat-module[]
+The `modules.d/beat-xpack.yml` file contains the following settings:
+
+[source,yaml]
+----------------------------------
+- module: beat
+  metricsets:
+    - stats
+    - state
+  period: 10s
+  hosts: ["http://localhost:5066"]
+  #username: "user"
+  #password: "secret"
+  xpack.enabled: true
+----------------------------------
```

Comment:
Other than mucking around with the `hosts`, `username`, and `password` setting, we really don't want users touching the other settings in this file. Is there some way we could achieve this in the docs, perhaps by not showing the entire config?

Reply:
I'd prefer to show the settings in context (otherwise, I worry that users might put the settings in the main config instead of the modules). However, I can add a blurb to indicate that users should accept defaults for other settings.